src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_PatternSwitchStatement.cs

```diff
@@ -402,6 +402,7 @@ private void LowerConstantValueDecision(DecisionTree.ByValue byValue)
                     {
                         var loweredRight = kv.Key;
                         var loweredLeft = kv.Value;
+                        loweredRight = _factory.Convert(loweredLeft.Type, loweredRight);
```

Comment:
You appear to be testing a version of the compiler that does not have the conversion inserted by this PR. We now insert a conversion in the assignment. There are scores of different kinds of conversions that the compiler is capable of producing, and unless you are questioning whether `_factory.Convert` can produce correct code for them, I do not think it makes sense to test them all for this particular context. I will add a couple of cases.